pkg/liquidity-mining/contracts/gauges/RewardsOnlyGauge.vy

```diff
@@ -0,0 +1,613 @@
+    """
+    @notice Get the number of decimals for this token
+    @dev Implemented as a view method to reduce gas costs
+    @return uint256 decimal places
+    """
+    return 18
+
+@view
+@internal
+def _get_allowance(owner: address, spender: address) -> uint256:
+    """
+     @dev Override to grant the Vault infinite allowance, causing for Gauge Tokens to not require approval.
+     This is sound as the Vault already provides authorization mechanisms when initiating token transfers, which this
+     contract inherits.
+    """
+    if (spender == BAL_VAULT):
+        return MAX_UINT256
+    return self._allowance[owner][spender]
+
+@internal
+def _checkpoint_rewards(_user: address, _total_supply: uint256, _claim: bool, _receiver: address):
+    """
+    @notice Claim pending rewards and checkpoint rewards for a user
+    """
+    # claim from reward contract
+
+    reward_data: uint256 = self._reward_data
+    if _total_supply != 0 and reward_data != 0 and block.timestamp > shift(reward_data, -160) + CLAIM_FREQUENCY:
+        reward_contract: address = convert(reward_data % 2**160, address)
+        raw_call(reward_contract, self.claim_sig)  # dev: bad claim sig
+        self._reward_data = convert(reward_contract, uint256) + shift(block.timestamp, 160)
+
+    receiver: address = _receiver
+    if _claim and receiver == ZERO_ADDRESS:
+        # if receiver is not explicitly declared, check for default receiver
+        receiver = self.rewards_receiver[_user]
+        if receiver == ZERO_ADDRESS:
+            # direct claims to user if no default receiver is set
+            receiver = _user
+
+    # calculate new user reward integral and transfer any owed rewards
+    user_balance: uint256 = self.balanceOf[_user]
+    for i in range(MAX_REWARDS):
+        token: address = self.reward_tokens[i]
+        if token == ZERO_ADDRESS:
+            break
+        dI: uint256 = 0
+        if _total_supply != 0:
+            token_balance: uint256 = ERC20(token).balanceOf(self)
+            dI = 10**18 * (token_balance - self.reward_balances[token]) / _total_supply
+            self.reward_balances[token] = token_balance
+            if _user == ZERO_ADDRESS:
+                if dI != 0:
+                    self.reward_integral[token] += dI
+                continue
+
+        integral: uint256 = self.reward_integral[token] + dI
+        if dI != 0:
+            self.reward_integral[token] = integral
+
+        integral_for: uint256 = self.reward_integral_for[token][_user]
+        new_claimable: uint256 = 0
+        if integral_for < integral:
+            self.reward_integral_for[token][_user] = integral
+            new_claimable = user_balance * (integral - integral_for) / 10**18
+
+        claim_data: uint256 = self.claim_data[_user][token]
+        total_claimable: uint256 = shift(claim_data, -128) + new_claimable
+        if total_claimable > 0:
+            total_claimed: uint256 = claim_data % 2 ** 128
+            if _claim:
```

Comment:
It seems there's a difference here between the deployed version I looked at and [the code in the Curve repo](https://github.com/curvefi/curve-dao-contracts/blob/master/contracts/gauges/RewardsOnlyGauge.vy):

![image](https://user-images.githubusercontent.com/2530770/161821080-5d287616-1a96-42bc-831d-913b820c6968.png)

The change was introduced here: https://github.com/curvefi/curve-dao-contracts/pull/105. It seems like it's the only difference between the code deployed and the code on the repo.

Reply:
The current implementation looks correct and more gas efficient so I'm inclined to leave it as is.